docs/book/checkout.rst

```diff
@@ -0,0 +1,259 @@
+
+    $stateMachine = $stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)
+    $stateMachine->apply(OrderCheckoutTransitions::TRANSITION_SELECT_SHIPPING);
+
+    $this->container->get('sylius.manager.order')->flush();
+
+**What happens with the transition?**
+
+The method ``recalculate($order)`` of the ``Sylius\Component\Core\OrderProcessing\OrderRecalculator`` is run.
+Here this method is responsible for: controlling the **shipping charges** which depend on the chosen ShippingMethod,
+controlling the **promotions** that depend on the shipping method.
+
+Selecting payment
+~~~~~~~~~~~~~~~~~
+
+This is a step where the customer chooses how are they willing to pay for their order.
+Basing on the PaymentMethods configured in the system the possibilities for the Customer are provided.
+
++----------------------------------------------+-------------------------------------------------+
+| Transition after step                        | Template                                        |
++----------------------------------------------+-------------------------------------------------+
+| ``shipping_selected``-> ``payment_selected`` | ``SyliusShopBundle:Checkout:payment.html.twig`` |
++----------------------------------------------+-------------------------------------------------+
+
+How to perform the Selecting payment step programmatically?
+'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Before this step your Order should be in the ``shipping_selected`` state. It will have a default Payment selected after the addressing step,
+but in this step you can change it.
+
+Firstly either create new (see how in the `Payments concept </book/payments`) or retrieve a **PaymentMethod**
+from the repository to assign it to your order's payment created defaultly in the addressing step.
+
+.. code-block:: php
+
+    // Let's assume that you have a method with code 'paypal' configured
+    $paymentMethod = $this->container->get('sylius.repository.payment_method')->findOneByCode('paypal');
+
+    // Payments are a Collection, so even though you hve one Payment by default you have to iterate over them
+    foreach ($order->getPayments() as $payment) {
+        $payment->setMethod($paymentMethod);
+    }
+
+After that get the StateMachine for the Order via the StateMachineFactory with a proper schema,
+and apply a proper transition and flush the order via the manager.
+
+.. code-block:: php
+
+    $stateMachineFactory = $this->container->get('sm.factory');
+
+    $stateMachine = $stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)
+    $stateMachine->apply(OrderCheckoutTransitions::TRANSITION_SELECT_PAYMENT);
+
+    $this->container->get('sylius.manager.order')->flush();
+
+**What happens with the transition?**
+
+The method ``recalculate($order)`` of the ``Sylius\Component\Core\OrderProcessing\OrderRecalculator`` is run.
+Here this method is responsible for controlling the **promotions** that depend on the payment method.
```

Comment:
Maybe we should mention something about adjustments, and what type of adjustment is created during checkout?